extensions/gitpod/src/collectLogs.ts

```diff
@@ -0,0 +1,95 @@
+import * as path from 'path';
+import * as yazl from 'yazl';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as vscode from 'vscode';
+
+interface IFile {
+    path: string;
+    contents: Buffer | string;
+}
+
+function zip(zipPath: string, files: IFile[]): Promise<string> {
+    return new Promise<string>((c, e) => {
+        const zip = new yazl.ZipFile();
+        files.forEach(f => {
+            if (f.contents) {
+                zip.addBuffer(typeof f.contents === 'string' ? Buffer.from(f.contents, 'utf8') : f.contents, f.path);
+            }
+        });
+        zip.end();
+
+        const zipStream = fs.createWriteStream(zipPath);
+        zip.outputStream.pipe(zipStream);
+
+        zip.outputStream.once('error', e);
+        zipStream.once('error', e);
+        zipStream.once('finish', () => c(zipPath));
+    });
+}
+
+async function traverseFolder(folderUri: vscode.Uri, files: IFile[], token: vscode.CancellationToken) {
+    if (token.isCancellationRequested) {
+        return;
+    }
+
+    const children = await vscode.workspace.fs.readDirectory(folderUri);
+    for (const [name, type] of children) {
+        if (token.isCancellationRequested) {
+            return;
+        }
+
+        if (type === vscode.FileType.File) {
+            const filePath = path.posix.join(folderUri.path, name);
+            const fileContent = await vscode.workspace.fs.readFile(folderUri.with({ path: filePath }));
+            if (fileContent.byteLength > 0) {
+                files.push({
+                    path: filePath,
+                    contents: Buffer.from(fileContent)
+                });
+            }
+        } else if (type === vscode.FileType.Directory) {
+            const folderPath = path.posix.join(folderUri.path, name);
+            await traverseFolder(folderUri.with({ path: folderPath }), files, token);
+        }
+    }
+}
+
+export async function collectLogs(context: vscode.ExtensionContext) {
+    const saveUri = await vscode.window.showSaveDialog({
+        title: 'Choose save location ...',
+        defaultUri: vscode.Uri.file(path.posix.join(os.homedir(), 'vscodeLogs.zip')),
+    });
+    if (!saveUri) {
+        return;
+    }
+
+    // Invoke command from gitpot-remote extension
+    const extRemoteLogsUri: vscode.Uri = await vscode.commands.executeCommand('__gitpod.getGitpodRemoteLogsPath');
```

Comment:
please handle a case when this command is missing for backward compatibility with old versions of gitpod-remote extension